source-academy/cadet-frontend
# Refactor Autograder and sub-components as function components

## Refactor Autograder and sub-components as function components
Summary: Code quality improvements and UX changes.

### Changelog
- Rename 'Other' assessment tab to 'Others' (following PR #1464)
- Refactor side-content Autograder as a function component using React hooks
  - Constant elements/values (independent of props) have been extracted out to avoid unnecessary computation
  - There is no change in functionality, except a minor performance improvement
- Refactor Autograder testcase and result cards as function components using React hooks
- Increase the duration of the notification shown when executing all testcases to 2000ms from 750ms
- Update affected Autograder snapshots

### Verification instructions
1. Configure the development front-end to use the mocks (set `REACT_APP_USE_BACKEND` to `FALSE` in `.env`)
2. Navigate to any assessment or submission, then proceed to any **programming** (not MCQ) question
3. Bring up the side-content Autograder (airplane icon), and verify there are no regressions
    - Verify that the testcase and autograding result sections in the Autograder can be independently collapsed
    - Click on a single testcase and verify that it runs
    - Evaluate the editor's contents (click the Run button or press Shift-Enter) and verify that all the testcases are run

Last updated 15 Aug 2020, 11:30PM
